### PR TITLE
Rework Select Sonar

### PR DESCRIPTION
**Pull request by ctuguinay**
The primary change is that the function now uses dataset/data array time variable start and end selection instead of filename time values.

### CODE REVIEW

**Comment by ctuguinay** [comment]
@leewujung This is ready to be reviewed

**Comment by ctuguinay** [comment]
Codecov says 'All modified and coverable lines are covered by tests ✅', so I think this coverage difference is alright?

**Review by leewujung — CHANGES_REQUESTED** [review]
Hey @ctuguinay : Thanks for the PR. Sorry it took a while to get to this. 

The modifications to the function looks great. The only thing I would suggest is to add a test to test for actually "gating" out the wrong Sv datasets/dataarrays completely outside of the regions start/end times.

The current test contains 1 dataset loaded from file and 1 empty dataset. How about constructing a few tiny datasets with just a few time points, and check if the right datasets are selected?

**Comment by ctuguinay** [comment]
Hey @leewujung, I added a few more checks for the gating in the `test_select_sonar` function. Do you think I should remove the `parse_simrad_fname_time` function from `utils/time.py`? The only place it was used was in the previous iteration of `select_sonar`.

**Comment by leewujung** [comment]
Yeah, I think that function could be removed.

**Comment by ctuguinay** [comment]
@leewujung Just removed it

File: echoregions/tests/test_regions2d.py [hunk 1]
```diff
@@ -249,9 +249,9 @@ def test_plot(regions2d_fixture: Regions2D) -> None:
 
 
 @pytest.mark.regions2d
-def test_select_sonar_file(regions2d_fixture: Regions2D, da_Sv_fixture: DataArray) -> None:
+def test_select_sonar(regions2d_fixture: Regions2D, da_Sv_fixture: DataArray) -> None:
     """
-    Tests select sonar file.
+    Tests select_sonar.
```
Comment (leewujung) [inline echoregions/tests/test_regions2d.py:254]: Maybe here these should be `select_sonar_file` to be consistent with the function name?

**Review by leewujung — APPROVED** [review]
Looks good! I only had a small comment of test function name. Feel free to merge after that.

**Comment by leewujung** [comment]
@ctuguinay : I pulled in all changes from `main`.

**Pull request merged**